rhino/src/main/java/org/mozilla/javascript/IdScriptableObject.java

```diff
@@ -936,48 +936,82 @@ protected ScriptableObject getOwnPropertyDescriptor(Context cx, Object id) {
         return desc;
     }
 
-    private ScriptableObject getBuiltInDescriptor(String name) {
-        Object value = null;
-        int attr = EMPTY;
+    private Slot queryOrFakeSlot(Context cx, Object id) {
+        var slot = querySlot(cx, id);
+        if (slot == null) {
+            if (id instanceof String) {
+                return getBuiltInSlot((String) id);
+            }
+
+            if (ScriptRuntime.isSymbol(id)) {
+                if (id instanceof SymbolKey) {
+                    return getBuiltInSlot((SymbolKey) id);
+                }
 
+                return getBuiltInSlot(((NativeSymbol) id).getKey());
+            }
+        }
+        return slot;
+    }
+
+    private ScriptableObject getBuiltInDescriptor(String name) {
```

Comment:
Maybe getBuiltInDataDescriptor is a better name

Reply:
Agreed, and changed.